test/ko_test.go

```diff
@@ -22,32 +22,15 @@ import (
 	"errors"
 	"fmt"
 	"os"
-	"testing"
 )
 
-var (
-	// Wether missing KO_DOCKER_REPO environment variable should be fatal or not
-	missingKoFatal = "true"
```

Comment:
This was put there, as is, to enable us to skip those on our OpenShift CI, while making it fail on "normal" CI.

Reply:
We could have a flag / environment variable about skipping vs failing in case of missing. 
I don't see the value of having it hardcoded in there?

Reply:
It's not "harcoded" per say as.. it's used with `-ldflags '-X github.com/tektoncd/pipeline/test/v1alpha1.missingKoFatal=false' `. An environment variable would work to I think.

Reply:
ok - I see. 

For the kaniko tests we do provide the option of setting up a registry in a sidecar, so a missing KO_DOCKER_REPO is not fatal. The way it is now the sidecar is always setup, but I think we should really do that only if no KO_DOCKER_REPO is provided. 

So, bottom line, I think the missing KO_DOCKER_REPO should never be globally "true", which is why I deleted it there.
Another way we could do this is to leave the two functions in place, and have the kaniko test use `GetDockerRepo` and the helm test use `EnsureDockerRepo`.

I'll make an update.